test/integration/targets/openssl_csr/tasks/main.yml

```diff
@@ -7,7 +7,8 @@
       openssl_csr:
         path: '{{ output_dir }}/csr.csr'
         privatekey_path: '{{ output_dir }}/privatekey.pem'
-        commonName: 'www.ansible.com'
+        subject:
```

Comment:
Could you leave one of the test with the previous interface, so we know it still does work. I've tried it locally and I know it works but so that test also shows (and ensure) that.

Reply:
I'd rather remove the previous interface completely, because it is redundant.

Reply:
Agree but we can't do that from 2.4 to 2.5. In order to remove it cleanly we need to print warning message "Usage of commonName is deprecated and will be dropped in 2.6, please use subject: commonName".

Else people with a working playbook on 2.4, when they will update to 2.5 will have a broken playbook. I need to check with the community how many release notice we should be warning (if 1 we can drop it of 2.6, if 2 then it will be dropped in 2.7)

Reply:
I mean, it states `This module is flagged as preview which means that it is not guaranteed to have a backwards compatible interface.`, but I agree that it would still be nice to at least have one version with a deprecation warning in the documentation.

Reply:
Yeah, but even for preview modules people expect certain consistency between N and N+1 if it can be done. Here the code is actually backward compatible. The comment was more regarding the test and ensuring while we guarantee it we still maintain the soon to be legacy interface.